packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -30,10 +30,11 @@ type PatchedFetcher = Fetcher & {
   readonly _nextOriginalFetch: Fetcher
 }
 
-function isPatchedFetch(
-  fetch: Fetcher | PatchedFetcher
-): fetch is PatchedFetcher {
-  return '__nextPatched' in fetch && fetch.__nextPatched === true
+export const NEXT_PATCH_SYMBOL = Symbol.for('next-patch')
+
+function isFetchPatched() {
+  // @ts-ignore
+  return globalThis[NEXT_PATCH_SYMBOL] === true
```

Comment:
You can use `Reflect.get()` not to introduce `@ts-ignore` here. It's a property getter that's type-oblivious. 

```suggestion
  return Reflect.get(globalThis, NEXT_PATCH_SYMBOL) === true
```